### PR TITLE
#156913229 Set up Heroku

### PR DESCRIPTION
**Pull request by alexmochu**
### What does this PR do?
Set up a production server for the app on Heroku
### Task to be completed
- Add Procfile which is a mechanism for declaring what commands are run by hc-bulls application’s dynos on the Heroku platform
- Install and configure (hc/settings.py) the application to use Django-Heroku library. This will automatically configure DATABASE_URL, ALLOWED_HOSTS, WhiteNoise (for static assets), Logging, and Heroku CI for our application.
### What are the relevant pivotal tracker stories?
[#156913229](https://www.pivotaltracker.com/story/show/156913229)

### CODE REVIEW

**Comment by jimmykamau** [comment]
Email seems to be broken

**Review by jimmykamau — CHANGES_REQUESTED** [review]
Email seems to be broken, I can't get a login email.

**Comment by alexmochu** [comment]
@jimmykamau Yes that seems to be broken. I will be creating a new branch/story and resolve this.

**Comment by jimmykamau** [comment]
What's the progress on this?

**Comment by alexmochu** [comment]
@jimmykamau I have resolved the merge conflicts. You can now merge the pull request. The only issue is that since my branch had no circle-ci and travis-ci integration are failing to pass their builds.

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (jimmykamau) [inline .vscode/settings.json]: This folder should also be in the .gitignore

File: hc/settings.py [hunk 2]
```diff
@@ -12,15 +12,18 @@
...
 import warnings
+import django_heroku
 
 BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 
 HOST = "localhost"
 SECRET_KEY = "---"
 DEBUG = True
-ALLOWED_HOSTS = []
+ALLOWED_HOSTS = ['https://hc-bulls.herokuapp.com/']
```
Comment (jimmykamau) [inline hc/settings.py]: Also add the staging URL

**Review by jimmykamau — CHANGES_REQUESTED** [review]
You should be able to merge changes in `develop` to your branch and resolve this

File: hc/settings.py [hunk 3]
```diff
@@ -12,15 +12,18 @@
...
 import warnings
+import django_heroku
 
 BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 
 HOST = "localhost"
 SECRET_KEY = "---"
 DEBUG = True
-ALLOWED_HOSTS = []
+ALLOWED_HOSTS = ['https://hc-bullss.herokuapp.com/']
```
Comment (jimmykamau) [inline hc/settings.py]: Also add the production URL

File: hc/settings.py [hunk 4]
```diff
@@ -145,6 +148,13 @@
 PUSHOVER_EMERGENCY_RETRY_DELAY = 300
 PUSHOVER_EMERGENCY_EXPIRATION = 86400
 
+EMAIL_HOST=os.environ['EMAIL_HOST']
+EMAIL_PORT=os.environ['EMAIL_PORT']
+EMAIL_HOST_USER=os.environ['EMAIL_HOST_USER']
+EMAIL_HOST_PASSWORD=os.environ['EMAIL_HOST_PASSWORD']
+EMAIL_USE_TLS=os.environ['EMAIL_USE_TLS']
+SITE_ROOT=os.environ['SITE_ROOT']
```
Comment (jimmykamau) [inline hc/settings.py]: You've redeclared this, check line 126. Also, have an `if` statement that checks if the `SITE_ROOT` is in the environment variables and if not, sets it to `http://localhost:8000`

File: .vscode/settings.json (hunk 1 above)
Comment (alexmochu) [reply .vscode/settings.json]: I have successfully added .vscode file directory inside .gitignore

File: hc/settings.py (hunk 4 above)
Comment (jimmykamau) [reply hc/settings.py]: Change this line to:
```
SITE_ROOT = os.environ.get("SITE_ROOT") or "http://localhost:8000"
```

**Review by jimmykamau — APPROVED** [review]
I've fixed the crashes

**Pull request merged**